airbyte-cdk/bulk/core/base/src/testFixtures/kotlin/io/airbyte/cdk/command/CliRunner.kt

```diff
@@ -33,54 +40,73 @@ data object CliRunner {
         config: ConfigurationJsonObjectBase? = null,
         catalog: ConfiguredAirbyteCatalog? = null,
         state: List<AirbyteStateMessage>? = null,
-    ): BufferingOutputConsumer =
+    ): BufferingOutputConsumer {
+        val out = TestOutputStream()
         runConnector(op, config, catalog, state) { args: Array<String> ->
-            AirbyteSourceRunner(args)
+            AirbyteSourceRunner(args, out.beanDefinition)
         }
+        return out.results
+    }
 
     /** Same as [runSource] but for destinations. */
     fun runDestination(
         op: String,
         config: ConfigurationJsonObjectBase? = null,
         catalog: ConfiguredAirbyteCatalog? = null,
         state: List<AirbyteStateMessage>? = null,
-    ): BufferingOutputConsumer =
+        inputStream: InputStream,
+    ): BufferingOutputConsumer {
```

Comment:
I might be misunderstanding - doesn't this still have the problem that it waits for the destination to terminate before returning `out.results`? i.e. we can't inspect the destination output messages while the connector is running

I think this method either needs to accept the TestOutputStream as a param, or it needs to exfiltrate `out` before invoking `runConnector`?

Reply:
you're right!